mito-ai/mito_ai/gemini_client.py

```diff
@@ -0,0 +1,70 @@
+# Copyright (c) Saga Inc.
+# Distributed under the terms of the GNU Affero General Public License v3.0 License.
+
+from typing import AsyncGenerator
+from google import genai
+from mito_ai.completions.models import AgentResponse
+
+
+class GeminiClient:
+    def __init__(self, api_key: str, model: str):
+        self.client = genai.Client(api_key=api_key)
+        self.model = model
+
+    async def generate_content(self, contents: str) -> str:
+        try:
+            response = self.client.models.generate_content(
+                model=self.model,
+                contents=contents,
+                config={
+                "response_mime_type": "application/json",
```

Comment:
The generate_content is used for inline code completion too. In that case we don't want to use AgentReponse format. Currently, we are getting this. 
 
<img width="477" alt="Screenshot 2025-05-19 at 10 54 36 AM" src="https://github.com/user-attachments/assets/db21b38d-4c3d-4133-9f91-0086d229ae87" />

You can see in the open ai that we are passing the response info from the specific prompt type completion_handler when we need it. We can use that again here. If the `ResponseFormatInfo` is provided, then we use that. Otherwise we don't set a response format.